File: upperbound.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int upperBound(vector<int> &arr, int x, int n)
{
    int low = 0, mid;
    int high = n - 1;
    int ans = n;
    while (low <= high)
    {
        mid = (low + high) / 2;
        if (arr[mid] > x)
        {
            ans = mid;
            high = mid - 1;
        }
        else
            low = mid + 1;
    }
    return ans;
}

File: UpperBound.java
public class UpperBound {

    public static int upperBound(int[] arr, int x, int n) {
        // Write your code here.
        int low = 0, high = n - 1;
        while (low <= high) {
            int mid = low + (high - low) / 2;
            if (arr[mid] <= x) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        return low;
    }

}